edge-modules/TwinTester/DesiredPropertyReceiver.cs

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace TwinTester
+{
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices;
+    using Microsoft.Azure.Devices.Client;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil;
+    using Microsoft.Azure.Devices.Shared;
+    using Microsoft.Extensions.Logging;
+
+    class DesiredPropertyReceiver : ITwinOperation
+    {
+        static readonly ILogger Logger = ModuleUtil.CreateLogger(nameof(DesiredPropertyReceiver));
+        readonly RegistryManager registryManager;
+        readonly ModuleClient moduleClient;
+        readonly ITwinTestResultHandler resultHandler;
+
+        public DesiredPropertyReceiver(RegistryManager registryManager, ModuleClient moduleClient, ITwinTestResultHandler resultHandler)
+        {
+            this.registryManager = registryManager;
+            this.moduleClient = moduleClient;
+            this.resultHandler = resultHandler;
+        }
+
+        public Task UpdateAsync()
```

Comment:
this looks a bit weird to me.  I expected this update method in DesiredPropertyReceiver is just called once ; but update method in DesiredPropertyUpdater, will call multiple times.  We can refactor in another PR.

Reply:
yes, it should be called once, I removed it from TwinAllOperations periodic because it was there by mistake

---
In reply to: [359095814](https://github.com/Azure/iotedge/pull/2108#discussion_r359095814) [](ancestors = 359095814)